c++/configure.ac

```diff
@@ -195,6 +200,19 @@ AS_IF([test "$with_openssl" != no], [
 ])
 AM_CONDITIONAL([BUILD_KJ_TLS], [test "$with_openssl" != no])
 
+AS_IF([test "$with_fibers" != no], [
+  CXXFLAGS="$CXXFLAGS -DKJ_USE_FIBERS"
+
+  AC_SEARCH_LIBS([getcontext], [], [], [
+    # If getcontext does not exist in libc, try to link libucontext
+    AC_CHECK_LIB(ucontext, getcontext, [
+      ASYNC_LIBS="$ASYNC_LIBS -lucontext"
+    ])
```

Comment:
I think we need a failure clause on this AC_CHECK_LIB to disable fibers? (Or error out if `--with-fibers` was specified explicitly.)

Reply:
Yep good point! However, I am not a big fan of changing the default based on what's found on the system. My proposition here is to error if the conditions are not met. The user can explicitly disable fibers in that case. That won't "just work", but that won't surprise anyone either.

What do you think?

Reply:
It's idiomatic with autotools to default to checking if the functionality is available and building it if it is. We already do that for zlib and openssl, for example. This is why the default for the `--with-x` flags is `check`, instead of `yes` or `no` -- `check` means "check if the library is available and use it if so".

If someone explicitly specifies `--with-fibers` (i.e. the value is `yes`, rather than `check`), and setcontext isn't available, then we should error out. But if they don't specify either `--with-fibers` or `--without-fibers` then we should enable or disable based on the result of the check.

(BTW, we might want to check for `makecontext`, `setcontext`, and/or `swapcontext` instead of `getcontext` -- I think some platforms have `getcontext` but don't support setting.)

Reply:
Right, I was not aware of that. I made it check for all 4 symbols (though KJ does not use `swapcontext` for what I could see). It is a bit verbose, but I don't think I can ask `AC_SEARCH_LIBS` or `AC_CHECK_LIB` to check a list of symbols.

Reply:
Oh, right, I think we used to use `swapcontext` but stopped because it wasn't supported somewhere... switched to just using `setcontext`.

I don't think it's necessary to check for all of them. The existence of either `makecontext` or `setcontext` probably implies the rest. I just meant that `getcontext` itself may not necessarily imply the others, so it's not the right one to check.

Reply:
Right :facepalm: I misunderstood that.